src/librbd/ImageCtx.cc

```diff
@@ -59,7 +59,7 @@ class ThreadPoolSingleton : public ThreadPool {
                  "rbd_op_threads") {
     start();
   }
-  virtual ~ThreadPoolSingleton() {
+  ~ThreadPoolSingleton() {
```

Comment:
my comment there suggested removing an empty destructor to dodge this issue :) i agree with @dillaman that we should stick to our style guide here